pkg/command/exec/exec.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// CombinedOutput runs the command 'cmd' which was initialized with reference
+// to 'ctx', and logs an error to 'scopedLog', with more verbosity if 'verbose'
+// is set to true.
+//
+// Returns any error (including timeout) that may have occurred during
+// execution of 'cmd'.
+func CombinedOutput(ctx context.Context, cmd *exec.Cmd, scopedLog *logrus.Entry, verbose bool) ([]byte, error) {
+	out, err := cmd.CombinedOutput()
+	if ctx.Err() == context.DeadlineExceeded {
+		scopedLog.WithField("cmd", cmd.Args).Error("Command execution failed: Timeout")
+		return nil, fmt.Errorf("Command execution failed: Timeout for %s", cmd.Args)
+	}
+	if err != nil {
+		if verbose {
```

Comment:
these two if clauses could be ANDed directly. We don't do anything in the inner else.